Description: Adapt libswmhack
 libswmhack tries to dlopen() libX11.so and libXt.so, but on Debian those
 are symlinks shipped in the corresponding -dev package.
 .
 If dlopen() fails, try with the versioned library names, libX11.so.6 and
 libXt.so.6. This ensures libswmhack will work with just the runtime
 packages installed.
Bug-Debian: https://bugs.debian.org/551732
Bug: https://github.com/conformal/spectrwm/pull/63
Author: Andrea Bolognani <eof@kiyuko.org>
Last-Update: 2018-08-09
Index: spectrwm/lib/swm_hack.c
===================================================================
--- spectrwm.orig/lib/swm_hack.c
+++ spectrwm/lib/swm_hack.c
@@ -79,6 +79,8 @@ get_atom_from_string(Display *dpy, char
 
 	if (lib_xlib == NULL)
 		lib_xlib = DLOPEN("libX11.so");
+	if (lib_xlib == NULL)
+		lib_xlib = DLOPEN("libX11.so.6");
 	if (lib_xlib) {
 		if (xia == NULL)
 			xia = (XIA *) dlsym(lib_xlib, "XInternAtom");
@@ -104,6 +106,8 @@ set_property(Display *dpy, Window id, At
 
 	if (lib_xlib == NULL)
 		lib_xlib = DLOPEN("libX11.so");
+	if (lib_xlib == NULL)
+		lib_xlib = DLOPEN("libX11.so.6");
 	if (lib_xlib) {
 		if (xcp == NULL)
 			xcp = (XCP *) dlsym(lib_xlib, "XChangeProperty");
@@ -131,6 +135,8 @@ XOpenDisplay(register _Xconst char *_dis
 
 	if (lib_xlib == NULL)
 		lib_xlib = DLOPEN("libX11.so");
+	if (lib_xlib == NULL)
+		lib_xlib = DLOPEN("libX11.so.6");
 	if (lib_xlib && func == NULL)
 		func = (ODF *) dlsym(lib_xlib, "XOpenDisplay");
 	if (func == NULL) {
@@ -167,6 +173,8 @@ XCreateWindow(Display *dpy, Window paren
 
 	if (lib_xlib == NULL)
 		lib_xlib = DLOPEN("libX11.so");
+	if (lib_xlib == NULL)
+		lib_xlib = DLOPEN("libX11.so.6");
 	if (lib_xlib && func == NULL)
 		func = (CWF *) dlsym(lib_xlib, "XCreateWindow");
 	if (func == NULL) {
@@ -206,6 +214,8 @@ XCreateSimpleWindow(Display *dpy, Window
 
 	if (lib_xlib == NULL)
 		lib_xlib = DLOPEN("libX11.so");
+	if (lib_xlib == NULL)
+		lib_xlib = DLOPEN("libX11.so.6");
 	if (lib_xlib && func == NULL)
 		func = (CSWF *) dlsym(lib_xlib, "XCreateSimpleWindow");
 	if (func == NULL) {
@@ -245,6 +255,8 @@ XtAppNextEvent(XtAppContext app_context,
 
 	if (lib_xtlib == NULL)
 		lib_xtlib = DLOPEN("libXt.so");
+	if (lib_xtlib == NULL)
+		lib_xtlib = DLOPEN("libXt.so.6");
 	if (lib_xtlib && func == NULL) {
 		func = (ANEF *) dlsym(lib_xtlib, "XtAppNextEvent");
 		if (display) {
